From 1c4e3e41d5245e5454449b01c12ed36f49a4007b Mon Sep 17 00:00:00 2001
From: Alok Meshram <meshram.alok@gmail.com>
Date: Wed, 29 May 2024 10:18:23 -0700
Subject: [PATCH 1/3] Explicitly specified C++17 for libslic3r and
 libslic3r_cgal in cmake . Also excluded template-id-cdtor from Werror. These
 changes fix compilation issues on Fedora 40.

---
 src/clipper2/CMakeLists.txt  | 2 +-
 src/libslic3r/CMakeLists.txt | 5 +++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/clipper2/CMakeLists.txt b/src/clipper2/CMakeLists.txt
index 275f9273b1..0dc24e5cac 100644
--- a/src/clipper2/CMakeLists.txt
+++ b/src/clipper2/CMakeLists.txt
@@ -37,7 +37,7 @@ target_include_directories(Clipper2
 if (WIN32)
   target_compile_options(Clipper2 PRIVATE /W4 /WX)
 else()
-  target_compile_options(Clipper2 PRIVATE -Wall -Wextra -Wpedantic -Werror)
+  target_compile_options(Clipper2 PRIVATE -Wall -Wextra -Wpedantic -Werror -Wno-error=template-id-cdtor)
   target_link_libraries(Clipper2 PUBLIC -lm)
 
 endif()
diff --git a/src/libslic3r/CMakeLists.txt b/src/libslic3r/CMakeLists.txt
index ccf30d8423..228e53ff70 100644
--- a/src/libslic3r/CMakeLists.txt
+++ b/src/libslic3r/CMakeLists.txt
@@ -456,6 +456,11 @@ if (_opts)
     target_compile_options(libslic3r_cgal PRIVATE "${_opts_bad}")
 endif()
 
+set_property(TARGET libslic3r PROPERTY CXX_STANDARD 17)
+set_property(TARGET libslic3r PROPERTY CXX_EXTENSIONS OFF)
+set_property(TARGET libslic3r_cgal PROPERTY CXX_STANDARD 17)
+set_property(TARGET libslic3r_cgal PROPERTY CXX_EXTENSIONS OFF)
+
 target_link_libraries(libslic3r_cgal PRIVATE ${_cgal_tgt} libigl mcut)
 
 if (MSVC AND "${CMAKE_SIZEOF_VOID_P}" STREQUAL "4") # 32 bit MSVC workaround

From d588f95198e152be2da1b1c032eb9023ce159939 Mon Sep 17 00:00:00 2001
From: Alok Meshram <meshram.alok@gmail.com>
Date: Wed, 29 May 2024 21:11:11 -0700
Subject: [PATCH 2/3] Only add -Wno-error=template-id-cdtor if gcc version is
 >= 14.1

---
 src/clipper2/CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/clipper2/CMakeLists.txt b/src/clipper2/CMakeLists.txt
index 0dc24e5cac..a9570bf0a9 100644
--- a/src/clipper2/CMakeLists.txt
+++ b/src/clipper2/CMakeLists.txt
@@ -37,9 +37,11 @@ target_include_directories(Clipper2
 if (WIN32)
   target_compile_options(Clipper2 PRIVATE /W4 /WX)
 else()
-  target_compile_options(Clipper2 PRIVATE -Wall -Wextra -Wpedantic -Werror -Wno-error=template-id-cdtor)
+  target_compile_options(Clipper2 PRIVATE -Wall -Wextra -Wpedantic -Werror)
   target_link_libraries(Clipper2 PUBLIC -lm)
-
+  if (CMAKE_COMPILER_IS_GNUCC AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 14.1)
+    target_compile_options(Clipper2 PRIVATE -Wno-error=template-id-cdtor)
+  endif()
 endif()
 
 set_target_properties(Clipper2 PROPERTIES FOLDER Libraries

From a27bf5f37e5a0ecfddaf16edb9ab2543662f5939 Mon Sep 17 00:00:00 2001
From: Alok Meshram <meshram.alok@gmail.com>
Date: Mon, 24 Jun 2024 19:56:19 -0700
Subject: [PATCH 3/3] Put -std=gnu++17 under gcc version >=14.1 to avoid
 compile-time error from earlier implementation of __int28

---
 src/libslic3r/CMakeLists.txt | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/libslic3r/CMakeLists.txt b/src/libslic3r/CMakeLists.txt
index 228e53ff70..eca70c4517 100644
--- a/src/libslic3r/CMakeLists.txt
+++ b/src/libslic3r/CMakeLists.txt
@@ -456,10 +456,12 @@ if (_opts)
     target_compile_options(libslic3r_cgal PRIVATE "${_opts_bad}")
 endif()
 
-set_property(TARGET libslic3r PROPERTY CXX_STANDARD 17)
-set_property(TARGET libslic3r PROPERTY CXX_EXTENSIONS OFF)
-set_property(TARGET libslic3r_cgal PROPERTY CXX_STANDARD 17)
-set_property(TARGET libslic3r_cgal PROPERTY CXX_EXTENSIONS OFF)
+if (CMAKE_COMPILER_IS_GNUCC AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 14.1)
+    set_property(TARGET libslic3r PROPERTY CXX_STANDARD 17)
+    set_property(TARGET libslic3r PROPERTY CXX_EXTENSIONS OFF)
+    set_property(TARGET libslic3r_cgal PROPERTY CXX_STANDARD 17)
+    set_property(TARGET libslic3r_cgal PROPERTY CXX_EXTENSIONS OFF)
+endif()
 
 target_link_libraries(libslic3r_cgal PRIVATE ${_cgal_tgt} libigl mcut)
 
